platform/lib/routers/pages.js

```diff
@@ -144,16 +149,41 @@ if (config.environment === 'development') {
 }
 
 if (config.environment !== 'development') {
-  pages.use('/', (request, response, next) => {
-    // Check if this request should be filtered
+  const STATIC_PAGES_PATH = utils.project.absolute('platform/pages');
+  const NOT_FOUND_ROUTE = '/404.html';
+
+  pages.use('/', async (request, response) => {
+    let requestPath = request.path;
+
+    // Match root requests to a possible index.html
+    if (requestPath.endsWith('/')) {
+      requestPath = requestPath + 'index.html';
+    }
+
+    // Check if the requested file exists
+    if (!fs.existsSync(utils.project.pagePath(requestPath))) {
+      requestPath = NOT_FOUND_ROUTE;
+      response.status(404)
+    }
+
+    let page = await readFileAsync(utils.project.pagePath(requestPath)).catch((e) => {
+      throw e;
```

Comment:
That's also something I am unsure about: my feeling is that this should somehow generate a 500 response, shouldn't it? I am curios what your thoughts are on this, @sebastianbenz.

Reply:
We should have a generic error handler (with a custom 500 page). See https://expressjs.com/en/guide/error-handling.html#writing-error-handlers

Reply:
`next(err)` instead of throw.